### PR TITLE
Ensure we don't return a nil control channel while starting up

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: router/router.go [hunk 1]
```diff
@@ -90,6 +90,10 @@ func (self *Router) MetricsRegistry() metrics.UsageRegistry {
 }
 
 func (self *Router) Channel() channel2.Channel {
+	// if we're just starting up, we may be nil. wait till initialized
+	for self.ctrl == nil {
```
Comment (ekoby) [inline router/router.go:94]: should we have some MAX time after which we stop the process?

Comment (plorenz) [reply router/router.go:94]: If the control channel isn't established within a certain amount of time, the router will exit

Comment (ekoby) [reply router/router.go:94]: some other code is handling that?

**Review by ekoby — APPROVED** [review]

File: router/router.go (hunk 1 above)
Comment (plorenz) [reply router/router.go:94]: will do. the initial control channel connect has a timeout

Comment (plorenz) [reply router/router.go:96]: Added comment

**Pull request merged**